exporter/prometheusexporter/prometheus.go

```diff
@@ -88,6 +91,6 @@ func (pe *prometheusExporter) ConsumeMetrics(_ context.Context, md pmetric.Metri
 	return nil
 }
 
-func (pe *prometheusExporter) Shutdown(context.Context) error {
-	return pe.shutdownFunc()
+func (pe *prometheusExporter) Shutdown(ctx context.Context) error {
+	return pe.shutdownFunc(ctx)
```

Comment:
nil check required

Reply:
Thanks for the hint. What would you prefer to happen when ctx is nil? Pass ctx.TODO on or return an error?

Reply:
Context should never be nil: https://pkg.go.dev/context#pkg-overview.

> Do not pass a nil [Context](https://pkg.go.dev/context#Context), even if a function permits it.

We also don't appear to check for nil context in any other components: https://github.com/search?q=repo%3Aopen-telemetry%2Fopentelemetry-collector-contrib+%22ctx+%3D%3D+nil%22&type=code

I would prefer keeping this as-is @atoulme